master/buildbot/newsfragments/git-submodules.bugfix

```diff
@@ -0,0 +1,2 @@
+Fixed bug with git checkouting from branch with submodule to branch without
```

Comment:
Users may be confused about what specific bug the release note refers to. I would suggest something like "Fixed :py:class:`~buildbot.steps.source.git.Git` to clean the repository after the checkout when submodules are enabled. Previously this action could lead to untracked module directories after changing branches".